src/htr2hpc/api_client.py

```diff
@@ -0,0 +1,329 @@
+            slugify(document_name).replace("-", "_")[:32],
+            file_format,
+            creation_time.strftime("%Y%m%d%H%M%S"),
+        )
+
+        return f"{self.base_url}/media/users/{user_id}/{base_filename}.zip"
+
+    def download_document_export(
+        self, user_id, document_id, document_name, transcription_ids
+    ) -> pathlib.Path:
+        """Request a document export, monitor the task until it completes,
+        then download the compiled document export file.
+        """
+
+        # use document_export api method to start an async export task
+        # NOTE: several more options should be configurable
+        result = self.document_export(
+            document_id=document_id,
+            transcription_id=transcription_ids,
+            include_images=True,
+        )
+        # currently returns status=ok, indicating a task has been queued
+        if result.status == "ok":
+            task_list = self.tasks()
+            # assume most recent task, which is listed first
+            # TODO: confirm correct task id based on method (export) and document name
+            export_task = task_list.results[0]
+
+        # until the task completes, sleep and then check status
+        # TODO: use task.workflow_state ?
+        # TODO: handle canceled task
+        while export_task.done_at is None:
+            sleep(1)
+            # refresh task details from api to check status
+            export_task = self.task(export_task.pk)
+
+        logger.info(f"Export XML completed after {export_task.duration()}")
+        export_file_url = self.export_file_url(
+            user_id, document_id, document_name, "alto", export_task.done_at
+        )
+        logger.info(f"Downloading export from {export_file_url}")
+
+        resp = requests.get(export_file_url, stream=True)
+        if resp.status_code == requests.codes.ok:
+            # content disposition includes filename
+            content_disposition = resp.headers.get("content-disposition")
+            content_length = resp.headers.get("content-length")
+            filename = content_disposition.split("filename=")[-1].strip('"')
+            outfile = pathlib.Path(filename)
+            # report on filename and size based on content-length header
+            logger.info(
+                f"Saving as {filename} ({humanize.naturalsize(content_length)})"
+            )
+            with outfile.open("wb") as filehandle:
+                for chunk in resp.iter_content(chunk_size=1024):
+                    filehandle.write(chunk)
+            return outfile
+
+    def tasks(self):
+        """paginated list of tasks"""
+        api_url = "tasks/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="task", **resp.json())
+
+    def task(self, task_id: int):
+        """details for a single task"""
+        api_url = f"tasks/{task_id}/"
+        resp = self._make_request(api_url)
+        return to_namedtuple("task", resp.json())
```

Comment:
We will also want functions to locate and download model files, if a user desires to train on top of a given model. For transcription, a user can either train on top of a given transcription model or from scratch. For segmentation, a user can either train on top of a given segmentation model or "from scratch", EXCEPT here "from scratch" actually means "on top of the base blla.mlmodel that comes as the default segmenter in kraken". So we need both a way of downloading a selected model and a way of passing the [blla.mlmodel](https://github.com/mittagessen/kraken/blob/main/kraken/blla.mlmodel) along with our data.

Fortunately, given a MODEL_PK, we can simply use BASE_URL/api/models/MODEL_PK/ and then get the URL to download the model in the "file" field. `requests.get()` should then work just fine.

For blla.mlmodel, we can just grab it from the kraken install, or store a copy somewhere.